python-sdk/src/langwatch/telemetry/context.py

```
@@ -122,12 +130,20 @@ def _reset_current_trace(token: contextvars.Token):
 
-def _reset_current_span(_token: str):
+def _reset_current_span(token):
     global main_thread_langwatch_span
     if not _is_on_child_thread():
         if len(main_thread_langwatch_span) > 0:
             main_thread_langwatch_span.pop()
 
+    if token is not None:
+        try:
+            stored_langwatch_span.reset(token)
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)